ghcide/src/Development/IDE/Core/FileStore.hs

```diff
@@ -127,16 +128,20 @@ getModificationTimeImpl vfs isWatched missingFileDiags file = do
         let file' = fromNormalizedFilePath file
         let wrap time@(l,s) = (Just $ LBS.toStrict $ B.encode time, ([], Just $ ModificationTime l s))
         mbVirtual <- liftIO $ getVirtualFile vfs $ filePathToUri' file
-        -- we use 'getVirtualFile' to discriminate FOIs so make that
-        -- dependency explicit by using the IsFileOfInterest rule
-        _ <- use_ IsFileOfInterest file
         case mbVirtual of
             Just (virtualFileVersion -> ver) -> do
                 alwaysRerun
                 pure (Just $ LBS.toStrict $ B.encode ver, ([], Just $ VFSVersion ver))
             Nothing -> do
                 isWF <- isWatched file
-                unless (isWF || isInterface file) alwaysRerun
+                unless (isWF || isInterface file)
```

Comment:
I find this logical really confusing. I think it's equivalent to:

```
if isInterfaceFile then do
    return () -- nothing to do
else if isWF then do
    alwaysRerun
else
    use_ IsFileOfInterest file
```

But I don't guarantee it - there's way too much negation, negation of negation etc.  Could we use that structure, and explain the three branches separately?

Reply:
I've done that, and it's much better now. Thanks!